src/platform/intel/ace/platform.c

```diff
@@ -59,15 +59,11 @@ int platform_boot_complete(uint32_t boot_message)
 	/* get any IPC specific boot message and optional data */
 	ipc_boot_complete_msg(&header, 0);
 
-	struct ipc_msg msg = {
-		.header = header.pri,
-		.extension = header.ext,
-		.tx_size = sizeof(ready),
-		.tx_data = (void *)&ready,
-	};
+	struct ipc_msg *msg = ipc_msg_init(header.pri, sizeof(header.pri));
 
 	/* send fimrware ready message. */
-	return ipc_platform_send_msg(&msg);
+	ipc_msg_send(ipc_msg, NULL, true);
+	return 0;
```

Comment:
Hmm, not following. You pass "ipc_msg" to ipc_msg_send() and msg is not used? And 'ready' is not used anywhere here. How can this work?

Reply:
Please have a look at function definition in src\ipc\ipc-common.c
![image](https://user-images.githubusercontent.com/12482515/194788645-881148ad-1751-4084-97a7-a84a59896e22.png)
@kv2019i please alaborate on what 'ready' is?
If you ask why `platform_boot_complete` function argument `uint32_t boot_message` is unused, it is because function signature must match generic for all platforms src\include\sof\platform.h declaration.
@softwarecki I'm glad your colleagues work puts you in a good mood.

Reply:
Ah I see now, was renaming variables really fast and forgot this one. Just a typo.